cmd/syncthing/gui.go

```diff
@@ -191,16 +191,18 @@ func (s *apiService) getListener(guiCfg config.GUIConfiguration) (net.Listener,
 		},
 	}
 
-	rawListener, err := net.Listen("tcp", guiCfg.Address())
-	if err != nil {
-		return nil, err
-	}
-
-	listener := &tlsutil.DowngradingListener{
-		Listener:  rawListener,
-		TLSConfig: tlsCfg,
+	listeners := make([]net.Listener, 0, len(guiCfgs))
+	for _, guiCfg := range guiCfgs {
+		rawListener, err := net.Listen("tcp", guiCfg.Address)
+		if err != nil {
+			return nil, err
+		}
+		listeners = append(listeners, &tlsutil.DowngradingListener{
```

Comment:
We should unwind and close all listeners in case of an error, or skip listeners that fail and abort if none exist.